src/Data/Text/Encoding.hs

```diff
@@ -443,6 +444,17 @@ encodeUtf8 (Text arr off len)
             memcpy ptr' ptr (fromIntegral utf8len)
             return (PS fp' 0 utf8len)
 
+-- | Encode text using ASCII encoding.
```

Comment:
What happens when text has characters outside the ASCII range?

Reply:
Then the output is garbage. We can call it `unsafeEncodeASCII` to be clearer.